frontend/libs/studio-components/src/components/StudioCodelistEditor/StudioCodeListEditorRow/StudioCodeListEditorRow.tsx

```diff
@@ -105,45 +105,141 @@ export function StudioCodeListEditorRow({
   );
 }
 
-type TextfieldCellProps = {
-  error?: string;
-  label: string;
-  onChange: (newString: string) => void;
+type InputCellProps = {
   value: CodeListItemValue;
+  label: string;
+  onChange: (newValue: CodeListItemValue) => void;
+  onFocus?: (event: FocusEvent) => void;
   autoComplete?: HTMLInputAutoCompleteAttribute;
+  error?: string;
 };
 
-function TextfieldCell({ error, label, value, onChange, autoComplete }: TextfieldCellProps) {
+function TypedInputCell({ error, label, value, onChange, autoComplete }: InputCellProps) {
   const ref = useRef<HTMLInputElement>(null);
 
   useEffect((): void => {
     ref.current?.setCustomValidity(error || '');
   }, [error]);
 
-  const handleChange = useCallback(
-    (event: React.ChangeEvent<HTMLInputElement>): void => {
-      onChange(event.target.value);
-    },
-    [onChange],
-  );
-
   const handleFocus = useCallback((event: FocusEvent<HTMLInputElement>): void => {
     event.target.reportValidity();
   }, []);
 
-  return (
-    <StudioInputTable.Cell.Textfield
-      aria-label={label}
-      autoComplete={autoComplete}
-      className={classes.textfieldCell}
-      onChange={handleChange}
-      onFocus={handleFocus}
-      ref={ref}
-      value={(value as string) ?? ''}
-    />
-  );
+  const shouldRenderNumberfield = typeof value === 'number';
+  const shouldRenderCheckbox = typeof value === 'boolean';
+
+  if (shouldRenderNumberfield) {
+    return (
+      <NumberfieldCell
+        label={label}
+        value={value}
+        autoComplete={autoComplete}
+        onChange={onChange}
+        onFocus={handleFocus}
+        ref={ref}
+      />
+    );
+  } else if (shouldRenderCheckbox) {
+    return (
+      <CheckboxCell
+        label={label}
+        value={value}
+        onChange={onChange}
+        onFocus={handleFocus}
+        ref={ref}
+      />
+    );
+  } else {
+    return (
+      <TextfieldCell
+        label={label}
+        value={value}
+        autoComplete={autoComplete}
+        onChange={onChange}
+        onFocus={handleFocus}
+        ref={ref}
+      />
+    );
+  }
 }
 
+const NumberfieldCell = forwardRef<HTMLInputElement, InputCellProps>(
+  ({ label, value, onChange, onFocus, autoComplete }, ref) => {
+    const handleNumberChange = useCallback(
+      (numberValue: number): void => {
+        if (numberValue === undefined) return;
```

Comment:
When will this case happen, and why is the check not present in the others? Should we consider using the same pattern as in the `StudioCodeListEditor` root - `isFieldValid(numberValue) && onChange(numberValue)`?